python/ml4ir/model/ranking_model.py

```diff
@@ -571,7 +567,9 @@ def _build_callback_hooks(
                 callbacks_list.append(cp_callback)
 
             # Early Stopping
-            early_stopping_callback = callbacks.EarlyStopping(monitor="val_new_MRR", mode="max")
+            early_stopping_callback = callbacks.EarlyStopping(
+                monitor="val_new_MRR", mode="max", patience=2, verbose=1, restore_best_weights=True
```

Comment:
yeah the "patience" param here is key when having `restore_best_weights` - because you let yourself overshoot.

Do you know if "patience" is measured in *epochs*?  If so, it this won't help much if we only do 3 epochs...

Reply:
`patience` is measured in `epochs`, yes. Should we make `patience` a function of `num_epochs`? Say 20%?

Reply:
I don't know if it needs to be an argument in itself.

Reply:
Also, FYI - I'm currently doing 10 epochs since I have all these built in.